climsight.py

```diff
@@ -383,6 +406,16 @@ def extract_climate_data(lat, lon, _hist, _future):
             soil = "Not known"
         st.markdown(f"**Soil type:** {soil}")
 
+        biodiversity = fetch_biodiversity(round(lat), round(lon))
+        biodiv_set = set()
+        for record in biodiversity['results']:
+            if record.get('taxonRank') != 'UNRANKED':
+                biodiv_set.add(record['genericName'])
+        biodiv = ', '.join(list(biodiv_set))
+        if biodiv == []:
+            biodiv = "Not known"
```

Comment:
Let's put all this logic into fetch_biodiversity function, so in this place you only get the text value, and display it.